### ISSUE TITLE
Reset accessibility for fields that are not accessible by default

### ISSUE THREAD

**Issue by jonas-grgt** [jonas-grgt/xjx#11]
When setting fields using reflection (ie. in `FieldReflector` ) check if the fields is accessible first, if not make it accessible and afterwards set the accessibility back to the default value.
